### PR TITLE
added tooltips to all node edit fields

### PR DESCRIPTION
**Pull request by FatherWolf**
added tooltip to every field in node edit to describe what it is used for

### CODE REVIEW

**Comment by FatherWolf** [comment]
Screenshot from 2022-07-01 15-13-09

File: src/i18n/resources/en.ts [hunk 1]
```diff
@@ -236,6 +236,43 @@ export const en: LanguageResource = {
...
+      'Unix timestamp of the last time the node checked in with the server. Used to determine generic health of node',
+    macaddress:
+      'he hardware mac address of the machine. Used to be used as the unique ID, but is being depreciated',
+    network: 'The network this node belongs too',
+    egressrange:
+      'If Egress is enabled, the gateway ranges that this machine routes to.',
+    nodeos: 'The operating system of the node',
+    nodelocalrange:
+      'If IsLocal has been enabled on the network, this is the local range in which the node will look for a private address from the local interfaces, to use as an endpoint',
+    isdnson: 'he node will setup DNS entries for its peers locally',
```
Comment (0xdcarns) [inline src/i18n/resources/en.ts]: missing a "T"

File: src/i18n/resources/en.ts [hunk 2]
```diff
@@ -236,6 +236,43 @@ export const en: LanguageResource = {
...
+    persistentkeepalive:
+      'How often packets are sent to keep connections open with other peers',
+    relayaddress:
+      'If “Relay” is enabled on this node, this field can be edited to add and remove nodes from the relay. So if you are currently relaying just one node but wish to relay an additional node, just add it’s private IP here',
+    nodeexpires:
+      'If a node should become invalid after a length of time, you can set it in this field, after which time, it will lose access to the network and will not populate to other nodes',
+    lastcheckin:
+      'Unix timestamp of the last time the node checked in with the server. Used to determine generic health of node',
+    macaddress:
+      'he hardware mac address of the machine. Used to be used as the unique ID, but is being depreciated',
```
Comment (0xdcarns) [inline src/i18n/resources/en.ts]: missing a "T"

Comment (0xdcarns) [reply src/i18n/resources/en.ts]: Also "Deprecated" not depreciated..

File: src/route/nodes/netid/nodeEdit/NodeEdit.tsx [hunk 3]
```diff
@@ -123,221 +121,275 @@ export const NodeEdit: React.FC<{
...
+        <Tooltip title={String(t('helper.dynamicendpoint'))} placement="top">
+          <Grid item xs={6} sm={4} md={3} sx={rowMargin}>
+            <NmFormInputSwitch
+              label={String(t('node.isstatic'))}
+              name={'isstatic'}
+              defaultValue={isIPDynamic}
+            />
+          </Grid>
+        </Tooltip>
+        <Tooltip title={String(t('helper.defaultlistenport'))} placement="top">
```
Comment (0xdcarns) [inline src/route/nodes/netid/nodeEdit/NodeEdit.tsx]: the tooltips should go around the NmFormInputTexts components.. if the fields are disabled, put a <span></span> around the input text first..

**Review by 0xdcarns — CHANGES_REQUESTED** [review]

File: src/route/nodes/netid/nodeEdit/NodeEdit.tsx (hunk 3 above)
Comment (FatherWolf) [reply src/route/nodes/netid/nodeEdit/NodeEdit.tsx]: done

File: src/route/nodes/netid/nodeEdit/NodeEdit.tsx [hunk 4]
```diff
@@ -121,26 +121,33 @@ export const NodeEdit: React.FC<{
...
+                defaultValue={String(node.listenport)}
+                name={'listenport'}
+                label={String(t('node.listenport'))}
+                type="number"
+                disabled={node.isserver || node.udpholepunch}
+              />
+            </span>
+          </Tooltip>
+        </Grid>
         <Tooltip title={String(t('helper.dynamicport'))} placement="top">
```
Comment (0xdcarns) [inline src/route/nodes/netid/nodeEdit/NodeEdit.tsx]: you can put this as the alternative to the !network?.defaultudpholepunch, then there won't be 2 tooltips..

**Review by 0xdcarns — CHANGES_REQUESTED** [review]

File: src/route/nodes/netid/nodeEdit/NodeEdit.tsx [hunk 5]
```diff
@@ -155,188 +168,263 @@ export const NodeEdit: React.FC<{
...
-              label={String(t('node.expdatetime'))}
-              value={
-                expTime
-                  ? datePickerConverter(expTime)
-                  : datePickerConverter(node.expdatetime)
-              }
-              onChange={(newValue: string | null) => {
-                if (!!newValue) {
-                  setExpTime(new Date(newValue).getTime() / 1000)
+        <Tooltip title={String(t('helper.nodeexpires'))} placement="top">
```
Comment (0xdcarns) [inline src/route/nodes/netid/nodeEdit/NodeEdit.tsx]: Any reason why this one is around the Grid item?

**Review by 0xdcarns — CHANGES_REQUESTED** [review]

**Review by 0xdcarns — APPROVED** [review]

**Pull request merged**